agentarchives/atom/client.py

```diff
@@ -0,0 +1,648 @@
+import json
+import logging
+import os
+import re
+import requests
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+__all__ = ['AtomError', 'ConnectionError', 'AuthenticationError', 'AtomClient']
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AtomError(Exception):
+    pass
+
+
+class ConnectionError(AtomError):
+    pass
+
+
+class AuthenticationError(AtomError):
+    pass
+
+
+class CommunicationError(AtomError):
+    def __init__(self, status_code, response):
+        message = "AtoM server responded {}".format(status_code)
+        self.response = response
+        super(CommunicationError, self).__init__(message)
+
+
+class AtomClient(object):
+    """
+    Client to communicate with a remote AtoM installation using its backend API.
+
+    Note that, while functions follow the same API as the ArchivistsToolkitClient, one major difference is the handling of resource and component IDs.
+    In ArchivistsToolkitClient, resource IDs are longs representing the database row ID.
+    In this client, resource IDs are instead slugs representing the unique part of a record's URL:
+        my-fonds-title
+    This change is due to the fact that slugs are visible by users whereas IDs aren't.
+    """
+
+    RESOURCE = 'resource'
+    RESOURCE_COMPONENT = 'resource_component'
+
+    def __init__(self, host, api_key, port=80):
+        parsed = urlparse(host)
+        if not parsed.scheme:
+            host = 'http://' + host
+
+        self.host = host + ':' + str(port)
+        self.api_key = api_key
+        self.api_base_url = 'ica-atom/index.php/api'
```

Comment:
We need to make the whole base URL customizable.
If we had `url` as suggested above and we received the following value: `https://archives.foobar.land/index.php` or even `https://archives.foobar.land` (when [no_script_enabled](http://symfony.com/legacy/doc/reference/1_4/en/04-Settings#chapter_04_sub_no_script_name) is enabled), you could do something like the following:

```python
try:
    from urllib.parse import urljoin
except ImportError:
    from urlparse import urljoin

base_api_url = urljoin('https://archives.foobar.land', 'api')
get_record_url = urljoin(base_api_url, 'informationobjects/{}'.format('foobar-slug'))

print(get_record_url)
```

Output: `https://archives.foobar.land/informationobjects/foobar-slug`